staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -89,9 +91,13 @@ type ServerRunOptions struct {
 	// This grace period is orthogonal to other grace periods, and
 	// it is not overridden by any other grace period.
 	ShutdownWatchTerminationGracePeriod time.Duration
+
+	// FeatureGate are the featuregate to install on the CLI
+	FeatureGate      featuregate.FeatureGate
+	EffectiveVersion utilversion.EffectiveVersion
 }
 
-func NewServerRunOptions() *ServerRunOptions {
+func NewServerRunOptions(featureGate featuregate.FeatureGate, effectiveVersion utilversion.EffectiveVersion) *ServerRunOptions {
```

Comment:
Please indicate that this is the version of the binary.

Now thinking about how this is used and the combination of this and CEL and knowing that CEL needs a kubeEffectiveVersion as opposed to a binaryEffectiveVersion, how do we expect a binary at version 4.18 to indicate a kubeEffectiveVersion at 1.31?

Reply:
why is this the binary version? It is whatever is passed to build the api server, which should probably be kubeEffectiveVersion.

Reply:
If I'm reading this comment thread, right, I think the ask is to add godoc along the lines of: 

```
// NewServerRunOptions creates and returns ServerRunOptions according to the given featureGate and effectiveVersion of the server binary to run. 
func NewServerRunOptions(featureGate featuregate.FeatureGate, effectiveVersion utilversion.EffectiveVersion) *ServerRunOptions
```

Reply:
Still missing godoc and it appears mis-wired.  I'm not seeing it coupled to the flags in here.